packages/strapi-provider-upload-aws-s3/lib/index.js

```diff
@@ -55,15 +55,15 @@ module.exports = {
   init: (config) => {
     // configure AWS S3 bucket connection
     AWS.config.update({
-      accessKeyId: config.public,
-      secretAccessKey: config.private,
+      accessKeyId: typeof config.public === "string" && config.public.trim(),
```

Comment:
Can you create a small util function to do this ;) 

```js
const trimParam = str => typeof str === "string" ? str.trim() : undefined`
```

And return undefined instead of false in case it's not a string :)